gamma_bridge/tasks.py

```diff
@@ -0,0 +1,33 @@
+"""Celery tasks for working asynchronously."""
+import socket
+
+from celery.task import task
+
+from gamma_bridge import settings
+from gamma_bridge.exceptions import GammaConnectionError
+from gamma_bridge.storage_logger import StorageLogger
+from gamma_bridge.storage import GammaStorage
+
+
+@task(bind=True)
+def publish_event_to_gamma_task(self, params, event, sleep_interval):
+    """
+    Send event to GammaStorage.
+    """
+    storage = GammaStorage(**params)
+    exception = None
+    try:
+        storage.save(event)
+    except socket.gaierror as e:  # can't connect at all, no response
+        exception = GammaConnectionError(message=event)
+
+    if storage.response_data is not None and not exception:
+        StorageLogger.logging(storage, event)
```

Comment:
Probably there can be two not handled exception from  StorageLogger:
```
NotBaseStorageError
GammaValueError
```

Reply:
I added the TODO comment to the StorageLogger: https://github.com/cmltaWt0/edx-gamma-bridge/pull/1/files#diff-7f44a7d7511cabbf240d08d5decc57b0R20 . It will be fixed in the future.